api/base/views.py

```diff
@@ -437,7 +460,27 @@ def root(request, format=None, **kwargs):
     if utils.has_admin_scope(request):
         return_val['meta']['admin'] = True
 
-    return Response(return_val)
+    resp = Response(return_val)
+
+    # This is used exclusively for our partnership with Sloan and can be deleted after their study is complete.
+    referer_url = request.environ.get('HTTP_REFERER', '')
+    if DOMAIN + 'preprints/' in referer_url:
+        provider = PreprintProvider.objects.get(_id=referer_url.split('/')[4])
+        if provider.in_sloan_study:
+            for key in sloan_data.keys():
+                set_tags_and_cookies_for_sloan(user, sloan_data[key], resp, key)
+
+    return resp
+
+
+def set_tags_and_cookies_for_sloan(user, value, resp, FLAG_NAME):
+    if user and not user.is_anonymous and not user.all_tags.filter(Q(name=FLAG_NAME) | Q(name='no_{}'.format(FLAG_NAME))):
+        if value:  # 50/50 chance flag is active
+            user.add_system_tag(FLAG_NAME)
+        else:
+            user.add_system_tag('no_{}'.format(FLAG_NAME))
+    resp.set_cookie(FLAG_NAME, value)
```

Comment:
What happens if the user has a system tag that differs from what the cookie had set? It looks like you would have a difference between the tag and the cookie, rather than overriding the cookie with what the user tag should be. Am I reading that right? It probably makes sense to keep the two consistent.

Reply:
It now overrides the cookie properly, includes a test.